File: JavaSolutions/IntersectionOfTwoArrays.java
import java.util.ArrayList;

class IntersectionOfTwoArrays {
    public int[] intersection(int[] nums1, int[] nums2) {
        ArrayList<Integer> al = this._intersectionHelper(nums1, nums2);
        int[] retArr = new int[al.size()];
        for (int i = 0; i < al.size(); i++) {
            retArr[i] = al.get(i);
        }
        return retArr;
    }

    private ArrayList<Integer> _intersectionHelper(int[] nums1, int[] nums2) {
        ArrayList retVal = new ArrayList<Integer>();
        for (int num : nums1) {
            for (int num2 : nums2) {
                if (num == num2 && !retVal.contains(num))
                    retVal.add(num);
            }
        }
        return retVal;
    }
}

File: PythonSolutions/IntersectionOfTwoArrays.py
from typing import List

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if len(nums1) <= len(nums2):
            return self._intersectionHelper(nums1, nums2)
        else:
            return self._intersectionHelper(nums2, nums1)

    def _intersectionHelper(self, nums1: List[int], nums2: List[int]) -> List[int]:
        """
        Precondtion: len(nums1) <= len(nums2)
        """
        ret_val = []
        for num in nums1:
            if num in nums2 and num not in ret_val:
                ret_val.append(num)

        return ret_val